src/app/clusters/basic-information/basic-information.cpp

```diff
@@ -343,6 +349,33 @@ CHIP_ERROR BasicAttrAccess::WriteLocation(AttributeValueDecoder & aDecoder)
     return DeviceLayer::ConfigurationMgr().StoreCountryCode(location.data(), location.size());
 }
 
+CHIP_ERROR BasicAttrAccess::ReadProductAppearance(AttributeValueEncoder & aEncoder)
+{
+    auto * provider = GetDeviceInstanceInfoProvider();
+    ProductFinishEnum finish;
+    ReturnErrorOnFailure(provider->GetProductFinish(&finish));
+
+    ColorEnum color;
+    CHIP_ERROR colorStatus = provider->GetProductPrimaryColor(&color);
+    if (colorStatus != CHIP_NO_ERROR && colorStatus != CHIP_ERROR_NOT_IMPLEMENTED)
+    {
+        return colorStatus;
+    }
```

Comment:
if colorStatus is NOT_IMPLEMENTED, should we explicitly set a default?

Reply:
No; this field is optional in the spec.